diff -Naur 001/po/cs.po 002/po/cs.po
--- 001/po/cs.po	2015-09-06 10:12:10.000000000 +0200
+++ 002/po/cs.po	2016-03-17 09:26:28.000000000 +0100
@@ -1524,7 +1524,7 @@
 #, c-format
 msgid "Server did not return any signed data for message ID `%s' on %s request"
 msgstr ""
-"Server nevrátil na požadavek %2$s žádná podepsaná data pro zprávu s ID „%1$s“"
+"Server nevrátil žádná podepsaná data pro zprávu s ID „%s“ na požadavek %s"
 
 #: src/isds.c:9822
 #, c-format
diff -Naur 001/src/isds.c 002/src/isds.c
--- 001/src/isds.c	2016-03-17 09:25:38.000000000 +0100
+++ 002/src/isds.c	2016-03-17 09:26:28.000000000 +0100
@@ -1356,7 +1356,7 @@
                             _("Selected authentication method: "
                                 "HMAC-based one-time password\n"));
                     authenticator_uri =
-                        "%1$sas/processLogin?type=hotp&uri=%1$sapps/";
+                        "%sas/processLogin?type=hotp&uri=%sapps/";
                     break;
                 case OTP_TIME: 
                     isds_log(ILF_SEC, ILL_INFO,
@@ -1368,16 +1368,16 @@
                                     "application, requesting server for "
                                     "new one.\n"));
                         authenticator_uri =
-                            "%1$sas/processLogin?type=totp&sendSms=true&"
-                            "uri=%1$sapps/";
+                            "%sas/processLogin?type=totp&sendSms=true&"
+                            "uri=%sapps/";
                     } else {
                         isds_log(ILF_SEC, ILL_INFO,
                                 _("OTP code has been provided by "
                                     "application, not requesting server "
                                     "for new one.\n"));
                         authenticator_uri =
-                            "%1$sas/processLogin?type=totp&"
-                            "uri=%1$sapps/";
+                            "%sas/processLogin?type=totp&"
+                            "uri=%sapps/";
                     }
                     break;
                 default:
@@ -1386,7 +1386,7 @@
                                 "method requested by application"));
                     return IE_ENUM;
             }
-            if (-1 == isds_asprintf(&context->url, authenticator_uri, url))
+            if (-1 == isds_asprintf(&context->url, authenticator_uri, url, url))
                 return IE_NOMEM; 
         }
     } else {
@@ -5330,7 +5330,7 @@
     {
         char *new_url = NULL;
         if ((err = _isds_build_url_from_context(context,
-                    "%1$.*2$sasws/changePassword", &new_url))) {
+                    "%.*sasws/changePassword", &new_url))) {
             goto leave;
         }
         saved_url = context->url;
@@ -5648,7 +5648,7 @@
         {
             char *new_url = NULL;
             if ((err = _isds_build_url_from_context(context,
-                        "%1$.*2$sasws/changePassword", &new_url))) {
+                        "%.*sasws/changePassword", &new_url))) {
                 goto leave;
             }
             saved_url = context->url;
diff -Naur 001/src/soap.c 002/src/soap.c
--- 001/src/soap.c	2016-03-17 09:25:38.000000000 +0100
+++ 002/src/soap.c	2016-03-17 09:26:28.000000000 +0100
@@ -1496,8 +1496,10 @@
 
 /* Build new URL from current @context and template.
  * @context is context carrying an URL
- * @template is printf(3) format string. First argument is string of base URL
- * found in @context, second argument is length of the base URL.
+ * @template is printf(3) format string. First argument is length of the base
+ * URL found in @context, second argument is the base URL, third argument is
+ * again the base URL.
+ * XXX: We cannot use "$" formatting character because it's not in the ISO C99.
  * @new_url is newly allocated URL built from @template. Caller must free it.
  * Return IE_SUCCESS, or corresponding error code and @new_url will not be
  * allocated.
@@ -1530,7 +1532,8 @@
     length++;
     
     /* Build new URL */
-    if (-1 == isds_asprintf(new_url, template, context->url, length))
+    if (-1 == isds_asprintf(new_url, template, length, context->url,
+                context->url))
         return IE_NOMEM;
 
     return IE_SUCCESS;
@@ -1551,7 +1554,7 @@
     /* Build logout URL */
     /*"https://DOMAINNAME/as/processLogout?uri=https://DOMAINNAME/apps/DS/WEB_SERVICE_ENDPOINT"*/
     err = _isds_build_url_from_context(context,
-            "%1$.*2$sas/processLogout?uri=%1$sDS/dz", &url);
+            "%.*sas/processLogout?uri=%sDS/dz", &url);
     if (err) return err;
 
     /* Invalidate the cookie by GET request */
